cli/azd/pkg/azapi/standard_deployments.go

```diff
@@ -688,3 +691,129 @@ func convertFromStandardProvisioningState(state armresources.ProvisioningState)
+		armresources.Deployment{
+			Properties: &armresources.DeploymentProperties{
+				Template:   armTemplate,
+				Parameters: parameters,
+				Mode:       to.Ptr(armresources.DeploymentModeIncremental),
+			},
+			Location: to.Ptr(location),
+			Tags:     tags,
+		}, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "subscription")
+	}
+
+	_, err = validate.PollUntilDone(ctx, nil)
+	if err != nil {
+		preflightError := createDeploymentError(err)
+		return fmt.Errorf(
+			"validating preflight to subscription:\n\nPreflight Error Details:\n%w",
+			preflightError,
+		)
+	}
+
+	return nil
+}
+
+type PreflightErrorResponse struct {
+	Error struct {
+		Code    string `json:"code"`
+		Message string `json:"message"`
+		Details []struct {
+			Code    string `json:"code"`
+			Message string `json:"message"`
+		} `json:"details"`
+	} `json:"error"`
+}
+
+func validatePreflightError(
+	rawResponse *http.Response,
+	err error,
+	typeMessage string,
+) error {
+	if rawResponse.StatusCode != 400 {
+		return fmt.Errorf("calling preflight validate api failing to %s: %w", typeMessage, err)
+	}
+
+	defer rawResponse.Body.Close()
+	body, errOnRawResponse := io.ReadAll(rawResponse.Body)
+	if errOnRawResponse != nil {
+		return fmt.Errorf("failed to read response error body from preflight api to %s: %w", typeMessage, errOnRawResponse)
+	}
+
+	var errPreflight PreflightErrorResponse
+	errOnRawResponse = json.Unmarshal(body, &errPreflight)
+	if errOnRawResponse != nil {
+		return fmt.Errorf("failed to unmarshal preflight error response to %s: %w", typeMessage, errOnRawResponse)
+	}
+
+	if len(errPreflight.Error.Details) > 0 {
+		detailMessage := errPreflight.Error.Details[0].Message
+		return fmt.Errorf("calling preflight validate api failing to %s: %s", typeMessage, detailMessage)
+	} else {
+		return fmt.Errorf("calling preflight validate api failing to %s: %w", typeMessage, err)
+	}
+}
+
+func (ds *StandardDeployments) ValidatePreflightToResourceGroup(
+	ctx context.Context,
+	subscriptionId, resourceGroup, deploymentName string,
+	armTemplate azure.RawArmTemplate,
+	parameters azure.ArmParameters,
+	tags map[string]*string,
+	options map[string]any,
+) error {
+	deploymentClient, err := ds.createDeploymentsClient(ctx, subscriptionId)
+	if err != nil {
+		return fmt.Errorf("creating deployments client: %w", err)
+	}
+
+	var rawResponse *http.Response
+	ctxWithResp := runtime.WithCaptureResponse(ctx, &rawResponse)
+
+	validate, err := deploymentClient.BeginValidate(ctxWithResp, resourceGroup, deploymentName,
+		armresources.Deployment{
+			Properties: &armresources.DeploymentProperties{
+				Template:   armTemplate,
+				Parameters: parameters,
+				Mode:       to.Ptr(armresources.DeploymentModeIncremental),
+			},
+			Tags: tags,
+		}, nil)
+	if err != nil {
+		return validatePreflightError(rawResponse, err, "resource group")
+	}
+
+	_, err = validate.PollUntilDone(ctx, nil)
+	if err != nil {
+		deploymentError := createDeploymentError(err)
```

Comment:
We should use this same `createDeploymentError()` func in the deployment stack use cases as well.